### PR TITLE
[Plugin] Explore features in Backstage

### PR DESCRIPTION
**Pull request by stefanalund**
## Hey, I just made a Pull Request!

Screenshot 2020-05-02 at 10 28 55

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/explore/src/components/ExplorePluginPage.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+const useStyles = makeStyles<BackstageTheme>(theme => ({
+  container: {
+    display: 'grid',
+    gridTemplateColumns: 'repeat(auto-fill, 296px)',
+    gridGap: theme.spacing(3),
+    marginBottom: theme.spacing(6),
+  },
+}));
+
+const toolsCards = [
```
Comment (stefanalund) [inline plugins/explore/src/components/ExplorePluginPage.tsx:39]: Ideally this data should be fetched directly from the plugins included in the Backstage deployment.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -4,4 +4,5 @@
 # The last matching pattern takes precedence.
 # https://help.github.com/articles/about-codeowners/
 
-* @spotify/backstage-core
+*                @spotify/backstage-core
+/plugins/explore @spotify/backstage-core
```
Comment (freben) [inline .github/CODEOWNERS:8]: We can remove these

File: plugins/explore/src/components/ExplorePluginPage.tsx (hunk 1 above)
Comment (Rugvip) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: With GitHub auth we could fetch info about existing plugins from GH, and maybe also include `[Plugin]` issues under an ideas/WIP section?

**Review by freben — APPROVED** [review]

File: yarn.lock [hunk 3]
```diff
@@ -16650,8 +16650,8 @@ promzard@^0.3.0:
 
 prop-types@^15.5.10, prop-types@^15.5.4, prop-types@^15.5.8, prop-types@^15.6.0, prop-types@^15.6.1, prop-types@^15.6.2, prop-types@^15.7.2:
   version "15.7.2"
-  resolved "https://registry.npmjs.org/prop-types/-/prop-types-15.7.2.tgz#52c41e75b8c87e72b9d9360e0206b99dcbffa6c5"
-  integrity sha512-8QQikdH7//R2vurIJSutZ1smHYTcLpRWEOlHnzcWHmBYrOGUysKwSsrC89BCiFj3CbrfJ/nXFdJepOVrY1GCHQ==
+  resolved "https://artifactory.spotify.net/artifactory/api/npm/virtual-npm/prop-types/-/prop-types-15.7.2.tgz#52c41e75b8c87e72b9d9360e0206b99dcbffa6c5"
+  integrity sha1-UsQedbjIfnK52TYOAga5ncv/psU=
```
Comment (freben) [inline yarn.lock:16654]: Oh no, wait, these are wrong

**Review by freben — CHANGES_REQUESTED** [review]

File: yarn.lock [hunk 4]
```diff
@@ -6472,8 +6472,8 @@ class-utils@^0.3.5:
 
 classnames@^2.2.5, classnames@^2.2.6:
   version "2.2.6"
-  resolved "https://registry.npmjs.org/classnames/-/classnames-2.2.6.tgz#43935bffdd291f326dad0a205309b38d00f650ce"
-  integrity sha512-JR/iSQOSt+LQIWwrwEzJ9uk0xfN3mTVYMwt1Ir5mUcSN6pU+V4zQFFaJsclJbPuAUQH+yfWef6tm7l1quW3C8Q==
+  resolved "https://artifactory.spotify.net/artifactory/api/npm/virtual-npm/classnames/-/classnames-2.2.6.tgz#43935bffdd291f326dad0a205309b38d00f650ce"
```
Comment (Rugvip) [inline yarn.lock:6475]: noooooope :p

Looks like you could just revert this change though, no new versions.

Still searching for a proper solution to this >_>

File: plugins/explore/src/components/ExploreCard.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  CardActions,
+  CardContent,
+  CardMedia,
+  Chip,
+  Typography,
+  makeStyles,
+} from '@material-ui/core';
+import { BackstageTheme } from '@backstage/theme';
+
+// import FollowNews from 'plugins/news/components/FollowNews';
```
Comment (Rugvip) [inline plugins/explore/src/components/ExploreCard.tsx:31]: Kill it with 🔥

File: plugins/explore/package.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@backstage/plugin-explore",
+  "version": "0.1.1-alpha.4",
+  "main": "dist/index.cjs.js",
```
Comment (Rugvip) [inline plugins/explore/package.json:4]: Should sync with master and change this to `dist/index.esm.js`

File: plugins/explore/src/components/ExplorePluginPage.tsx (hunk 1 above)
Comment (freben) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: Depends on what the purpose of this plugin is i guess. Who will consume the views? I thought this was for end users to easily find functionality and features.

Comment (Rugvip) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: We should def get backstage instance for managing this project up at some point, and this could be a part of that.

Comment (freben) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: So meta.

Yes that instance could even have a marketplace of sorts, but it would arguably be a different plugin than this one.

File: yarn.lock (hunk 4 above)
Comment (freben) [reply yarn.lock:6531]: There seem to exist some lock file linters. Let's see if one supports limiting the resolved field soon

Candidate: https://github.com/lirantal/lockfile-lint/blob/master/README.md

File: plugins/explore/src/components/ExplorePluginPage.tsx (hunk 1 above)
Comment (Rugvip) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: #697 

I don't see why we wouldn't want to promote WIP plugins at this point though, as in just include those in the exploration too.

File: yarn.lock (hunk 4 above)
Comment (Rugvip) [reply yarn.lock:6531]: Yeah it's a bit surprising that the build goes through

File: plugins/explore/src/components/ExplorePluginPage.tsx (hunk 1 above)
Comment (freben) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: I'm probably misunderstanding you, sorry for any confusion. I was just thinking that wip plugin promotion may not be helpful for potentially hundreds of thousands (dream big!) of end user Backstage employees out in the world, that have nothing to do with the actual implementation of the system internally.

Comment (stefanalund) [reply plugins/explore/src/components/ExplorePluginPage.tsx:39]: I see different uses of this depending on: 

Production - only plugins in your deployment of Backstage shows up. This feature helps users of Backstage find tools available. 

Dev - All open sourced plugins appear, helping Operators see what plugins are available. 

I think we should have a demo Dev instance at some point. Ping @decause

File: plugins/explore/src/components/ExploreCard.tsx (hunk 5 above)
Comment (stefanalund) [reply plugins/explore/src/components/ExploreCard.tsx:31]: Gone.

File: .github/CODEOWNERS (hunk 2 above)
Comment (stefanalund) [reply .github/CODEOWNERS]: Gone.

**Comment by stefanalund** [comment]
@freben @Rugvip I deleted `yarn.lock` and re-ran `yarn install`. The resulting file is very different from the one checked in. Any suggestions for how to proceed?

**Comment by freben** [comment]
@stefanalund yeah that re-evaluates every single dependency. What you want to do is to grab the original file from master. `git checkout master -- yarn.lock` in the root (and then committing that with the rest, since that's a diff from what you had so far) should do it

**Comment by stefanalund** [comment]
@freben fixed! Thanks for help.

File: plugins/explore/src/components/ExploreCard.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            <Chip
+              label={lifecycle}
+              className={classNames(
+                classes.lifecycle,
+                classes[lifecycle.toLowerCase()],
+              )}
+            />
+          )}
+        </Typography>
+        <Typography component="p">
```
Comment (freben) [inline plugins/explore/src/components/ExploreCard.tsx:110]: There's `<Typography paragraph>` for this

**Review by freben — APPROVED** [review]
Looking good! Just minor nit, 🚢
